pyvista/utilities/geometric_objects.py

```diff
@@ -1264,9 +1268,82 @@ def Rectangle(points=None):
     >>> pointa = [1.0, 0.0, 0.0]
     >>> pointb = [1.0, 1.0, 0.0]
     >>> pointc = [0.0, 1.0, 0.0]
-    >>> pointd = [0.0, 0.0, 0.0]
-    >>> rectangle = pyvista.Rectangle([pointa, pointb, pointc, pointd])
+    >>> rectangle = pyvista.Rectangle([pointa, pointb, pointc])
     >>> rectangle.plot(show_edges=True, line_width=5)
+    """
+    if points is None:
+        points = [[1.0, 0.0, 0.0], [1.0, 1.0, 0.0], [0.0, 1.0, 0.0]]
+    if len(points) == 4:
+        warnings.warn(
+            'Rectangle defined by 4 points is deprecated. Please use ``pyvista.Quadrilateral``.',
+            PyVistaDeprecationWarning,
+        )
+        return Quadrilateral(points)
+    elif len(points) != 3:
+        raise TypeError('Points must be given as length 3 np.ndarray or list')
+
+    check_valid_vector(points[0], 'points[0]')
+    check_valid_vector(points[1], 'points[1]')
+    check_valid_vector(points[2], 'points[2]')
+
+    point_0 = np.array(points[0])
+    point_1 = np.array(points[1])
+    point_2 = np.array(points[2])
+
+    vec_01 = point_1 - point_0
+    vec_02 = point_2 - point_0
+    vec_12 = point_2 - point_1
+
+    scalar_pdct_01_02 = np.dot(vec_01, vec_02)
+    scalar_pdct_01_12 = np.dot(vec_01, vec_12)
+    scalar_pdct_02_12 = np.dot(vec_02, vec_12)
+
+    non_null_scalar_products = [
+        val
+        for val in [scalar_pdct_01_02, scalar_pdct_01_12, scalar_pdct_02_12]
+        if np.isclose(val, 0)
+    ]
+    if len(non_null_scalar_products) == 0:
+        raise ValueError("The three points should defined orthogonal vectors")
+    if len(non_null_scalar_products) > 1:
+        raise ValueError("Unable to build a rectangle with less than three different points")
+
+    points = np.array([point_0, point_1, point_2, point_0])
+    if np.isclose(scalar_pdct_01_02, 0):
+        points[3] = point_0 + vec_01 + vec_02
+        cells = np.array([[4, 0, 1, 3, 2]])
+    if np.isclose(scalar_pdct_01_12, 0):
+        points[3] = point_1 + vec_12 - vec_01
+        cells = np.array([[4, 0, 1, 2, 3]])
+    if np.isclose(scalar_pdct_02_12, 0):
+        points[3] = point_2 - vec_02 - vec_12
+        cells = np.array([[4, 0, 2, 1, 3]])
+
+    return pyvista.wrap(pyvista.PolyData(points, cells))
```

Comment:
Why are we wrapping PolyData?

Reply:
In fact i just took what existed in the former Rectangle function. This also what is done in the other function such a Triangle, Circle and Ellipse.

Reply:
This seems to be an existing redundant code. I decided to remove the other parts with https://github.com/pyvista/pyvista/pull/4146.
```suggestion
    return pyvista.PolyData(points, cells)
```

Reply:
See discussion in #4146.  I'd recommend reverting back to the original state of hte PR and any cleanup could be done in #4146 if we decide to go in a different direction.